common/flags/index.ts

```diff
@@ -0,0 +1,128 @@
+import Long from "long";
+
+export class Flags {
+  public static boolean(name: string, defaultValue: boolean = false) {
+    return new SingleValueFlag(name, defaultValue, (stringValue: string) => {
+      if (stringValue === "" || stringValue === "true") {
+        return true;
+      }
+      if (stringValue === "false") {
+        return false;
+      }
+      throw Flags.invalidFlagValueError(name);
+    });
+  }
+
+  public static string(name: string, defaultValue: string = "") {
+    return new SingleValueFlag(name, defaultValue, (stringValue: string) => stringValue);
+  }
+
+  public static number(name: string, defaultValue: number = 0) {
+    return new SingleValueFlag(name, defaultValue, (stringValue: string) =>
+      parseFloat(stringValue)
+    );
+  }
+
+  public static long(name: string, defaultValue: Long = Long.ZERO) {
+    return new SingleValueFlag(name, defaultValue, Long.fromString);
+  }
+
+  public static stringSet(name: string, defaultValue: Set<string> = new Set()) {
+    return new SetFlag(name, defaultValue, (singleValue: string) => singleValue);
+  }
+
+  public static getRawFlagValue(flagName: string) {
+    return Flags.parsedArgv[flagName];
+  }
+
+  public static setRawFlagValueForTesting(flagName: string, value: string) {
+    Flags.parsedArgv[flagName] = value;
+  }
+
+  private static readonly parsedArgv = (() => {
```

Comment:
What happens if this throws? Will it just immediately kill the process? Assume that's what we want it to, just hope it doesn't manifest as an uncaught exception or something weird

Reply:
Yup, just tested, it does immediately kill the process.